Input/ClickHandler.gd

```diff
@@ -49,5 +56,22 @@ func _on_mouse_entered() -> void:
 func _on_mouse_exited() -> void:
 	if not _is_interactable:
 		return
-		
+	
+	_set_lock_hover_timer()
+	
 	on_unhover.emit()
+
+
+func _set_lock_hover_timer():
+	if lock_hover_time <= 0.0:
+		return
+	
+	if _lock_hover_timer != null:
+		_lock_hover_timer.stop()
+	
+	# Disable hovering, set timer. When timer expires, re-enable hovering
+	_hover_enabled = false
+	_lock_hover_timer = Timer.new()
+	add_child(_lock_hover_timer)
+	_lock_hover_timer.timeout.connect(func(): _hover_enabled = true)
```

Comment:
Want to sanity check the logic, are we okay with calling the connect here since this will be called everytime `_on_mouse_exited()` is triggered? Not sure where else would be a better place to call the connect function.

Reply:
That's a good point, it may be connecting multiple times? Perhaps it's a good idea to call `disconnect` before connecting again

Reply:
Or maybe we create the timer in _ready instead